src/Services/FHIR/FhirValueSetService.php

```diff
@@ -0,0 +1,207 @@
+        ];
+    }
+
+    /**
+     * Retrieves all of the fhir observation resources mapped to the underlying openemr data elements.
+     * @param $fhirSearchParameters The FHIR resource search parameters
+     * @return processing result
+     */
+    public function getAll($fhirSearchParameters, $puuidBind = null): ProcessingResult
+    {
+        $fhirSearchResult = new ProcessingResult();
+        try {
+            if (
+                !isset($fhirSearchParameters[ '_id' ])
+                // could be array (AND) or comma-delimited string value (OR)
+                // check array first but should only be len 1 ("AND", becuase cannot be 2 simultaneous)
+                || ( is_array($fhirSearchParameters[ '_id' ])
+                   && count($fhirSearchParameters[ '_id' ]) == 1
+                   && $fhirSearchParameters[ '_id' ][ 0 ] == self::APPOINTMENT_TYPE )
+                // and string which could be comma-delimiter OR of exploded values
+                || ( !is_array($fhirSearchParameters[ '_id' ])
+                   && in_array(self::APPOINTMENT_TYPE, explode(",", $fhirSearchParameters[ '_id' ])) )
+            ) {
+                $calendarCategories = $this->appointmentService->getCalendarCategories();
+                $valueSet = new FHIRValueSet();
+                $valueSet->setId(self::APPOINTMENT_TYPE);
+                $compose = new FHIRValueSetCompose();
+                $include = new FHIRValueSetInclude();
+                foreach ($calendarCategories as $category) {
+                    if ($category["pc_cattype"] != 0) {
+                        continue; // only cat_type==0
+                    }
+                    $concept = new FHIRValueSetConcept();
+                    $code = new FHIRCode();
+                    $code->setValue($category[ "pc_constant_id"]);
+                    $concept->setCode($code);
+                    $concept->setDisplay($category[ "pc_catname" ]);
+                    $include->addConcept($concept);
+                }
+                $compose->addInclude($include);
+                $valueSet->setCompose($compose);
+                $fhirSearchResult->addData($valueSet);
+            }
+
+              // Now the same for list_options selected in $listNames
+              $list_ids = $this->listOptionService->getListIds();
+            foreach ($list_ids as $listName) {
+                if (
+                    isset($fhirSearchParameters[ '_id' ])
+                    // could be array (AND) or comma-delimited string value (OR)
+                    // check array first but should only be len 1 ("AND", becuase cannot be 2 simultaneous)
+                    && ( ( is_array($fhirSearchParameters[ '_id' ])
+                         && count($fhirSearchParameters[ '_id' ]) == 1
+                         && $fhirSearchParameters[ '_id' ][ 0 ] == $listName )
```

Comment:
Shouldn't this be a ```$fhirSearchParameters[ '_id' ][ 0 ] != $listname``` in order to skip it if there is only one _id clause and it doesn't match the current list item? Or am I misreading this expression?

Reply:
ha ha yes